anta/tests/hardware.py

```diff
@@ -379,3 +379,58 @@ def test(self) -> None:
         # Verify that the expected redundancy protocol configured, operational and switchover ready
         elif not command_output["switchoverReady"]:
             self.result.is_failure(f"Redundancy protocol switchover status mismatch - Expected: True Actual: {command_output['switchoverReady']}")
+
+
+class VerifyTridentCounters(AntaTest):
+    """Verifies the trident counters for all interfaces.
+
+    Compatible with EOS operating in `trident` platform.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all interfaces have drop and error counter values below the drop threshold.
+    * Failure: The test will fail if any interface has drop or error counter values above the drop threshold.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.hardware:
+      - VerifyTridentCounters:
+    ```
+    """
+
+    categories: ClassVar[list[str]] = ["hardware"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show platform trident counters", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyTridentCounters test."""
+
+        drop_threshold: PositiveInteger = 0
+        """The upper limit for the configured packet drops or error counters."""
+
+    @skip_on_platforms(["cEOSLab", "vEOS-lab", "cEOSCloudLab", "vEOS"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyTridentCounters."""
+        self.result.is_success()
+        command_output = self.instance_commands[0].json_output
+
+        for interface, hw_counters in command_output["ethernet"].items():
+            for counter_name, drop_counter in hw_counters["count"]["drop"].items():
+                if counter_name in {"nonCongestionDiscard", "rxFpDrop"}:
+                    continue
+
+                # Verify actual drop threshold
+                if drop_counter > self.inputs.drop_threshold:
+                    self.result.is_failure(
+                        f"Interface: {interface} Drop Counter: {counter_name} - Drop threshold mismatch - Expected < {self.inputs.drop_threshold} "
+                        f"Actual: {drop_counter}"
+                    )
+
+            for counter_name, error_counter in hw_counters["count"]["error"].items():
+                # Verify actual error threshold
+                if counter_name != "rxVlanDrop" and error_counter > self.inputs.drop_threshold:
+                    self.result.is_failure(
+                        f"Interface: {interface} Error Counter: {counter_name} - Error threshold mismatch - Expected < {self.inputs.drop_threshold} "
```

Comment:
Same changes as above.
```suggestion
                        f"Interface: {interface} Error Counter: {counter_name} - Threshold exceeded - Expected: < {self.inputs.drop_threshold} "
```